### PR TITLE
Org admin page

### PR DESCRIPTION
**Pull request by ryanpbrewster**
@leogrim @andrewconner 

Not all the info is actually displayed on the pages, but it's all there. Still need to do Javascript magic to let admins set things like the org name, add candidate members, invite members, transfer ownership, etc.

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/admin/AdminUserController.scala [hunk 1]
```diff
@@ -9,7 +9,7 @@ import scala.concurrent.{ Await, Future, Promise }
 import scala.concurrent.duration.{ Duration, DurationInt }
 import scala.util.{ Failure, Success, Try }
 
-import com.google.inject.Inject
+import com.google.inject.{ Inject, Singleton }
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminUserController.scala:12]: Are controllers supposed to be Singletons?

File: kifi-backend/modules/shoebox/conf/shoeboxService.routes [hunk 2]
```diff
@@ -675,6 +675,11 @@ GET    /admin/users/mixpanel/delete          @com.keepit.controllers.admin.Admin
 POST    /admin/user/:id/emails/:code/send    @com.keepit.controllers.admin.AdminUserController.sendEmail(id: Id[User], code: String)
 POST    /admin/user/:id/refreshRecos    @com.keepit.controllers.admin.AdminUserController.refreshRecos(id: Id[User])
 
+
+GET     /admin/populatedb             @com.keepit.controllers.admin.AdminOrganizationController.populateDatabase()
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/conf/shoeboxService.routes]: I am considering making a real route like this. I found it extremely useful for testing the admin HTML pages. Thoughts?

File: kifi-backend/modules/shoebox/app/com/keepit/admin/AdminOrganizationController.scala [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    orgMembershipCandidateRepo: OrganizationMembershipCandidateRepo,
+    orgCommander: OrganizationCommander,
+    orgMembershipCommander: OrganizationMembershipCommander,
+    organizationMembershipCandidateCommander: OrganizationMembershipCandidateCommander,
+    organizationInviteCommander: OrganizationInviteCommander) extends AdminUserActions {
+
+  def populateDatabase() = AdminUserPage { implicit request =>
+    val ryan = userCommander.createUser("Ryan", "Brewster", addrOpt = None, state = UserStates.ACTIVE)
+    val users = for (i <- 1 to 20) yield {
+      userCommander.createUser(i.toString, "TestUser", addrOpt = None, state = UserStates.ACTIVE)
```
Comment (yingjieMiao) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminOrganizationController.scala]: unit test/dev mode is more appropriate for this type of thing. Also, users created this way are not marked as fake users.

Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminOrganizationController.scala]: Agreed, this seems dangerous, you could call this endpoint in production and create a bunch of fake data.

File: kifi-backend/modules/shoebox/app/com/keepit/admin/AdminUserController.scala (hunk 1 above)
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminUserController.scala:12]: I don't think so, but I don't remember if there's a good reason for them not to be @andrewconner

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala [hunk 4]
```diff
@@ -35,24 +42,33 @@ class OrganizationCommanderImpl @Inject() (
...
 
-      val avatarPath = organizationAvatarCommander.getBestImage(orgId, ImageSize(200, 200)).map(_.imagePath)
+    val avatarPath = organizationAvatarCommander.getBestImage(orgId, ImageSize(200, 200)).map(_.imagePath)
 
-      // TODO: actually find the number of libraries
-      val numPublicLibs = libraries(LibraryVisibility.PUBLISHED)
-      OrganizationView(orgId = orgId, handle = orgHandle, name = orgName, description = description, avatarPath = avatarPath, members = members,
-        numMembers = memberCount, numLibraries = numPublicLibs)
-    }
+    // TODO: actually find the number of libraries
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala]: Even better, find the number of libraries for each `LibraryVisibility` (hiding secret ones like we do for keeps), this should help us understand organization's main use cases for libraries. I can see Colin already introduced a `LibraryRepo.countLibrariesForOrgByVisibility`

Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala]: Whoops, this is already what you're using and I thought this was `Admin`. Here you're right, you want to personalize this count for the viewer :)

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala [hunk 5]
```diff
@@ -210,4 +221,39 @@ class OrganizationCommanderImpl @Inject() (
       }
     }
   }
+
+  // For use in the Admin Organization controller. Don't use it elsewhere.
+  def unsafeModifyOrganization(orgId: Id[Organization], modifications: OrganizationModifications): Unit = {
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala]: What about passing in the requester's info and double check for `Admin` experiment?

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala [hunk 6]
```diff
@@ -210,4 +221,39 @@ class OrganizationCommanderImpl @Inject() (
...
+        val memberships = orgMembershipRepo.getAllByOrgId(org.id.get)
+        applyNewBasePermissionsToMembers(memberships, org.basePermissions, modifiedOrg.basePermissions)
+      }
+    }
+  }
+
+  def getAnalyticsView(orgId: Id[Organization]): AnalyticsOrganizationView = {
+    db.readOnlyReplica { implicit session =>
+      val orgView = getOrganizationViewHelper(orgId)
+      // TODO: get actual numbers
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala]: Note: we have the convention of adding our name to todo's, e.g. `Todo(Léo)`. Increases responsibility for authors and traceability for readers :)

Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala]: That's a good idea.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala (hunk 5 above)
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala]: :+1:

**Pull request merged**